File: src/main/java/com/hjj/apiserver/domain/DepositPK.java
package com.hjj.apiserver.domain;

import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@NoArgsConstructor
public class DepositPK implements Serializable {

    private static final long serialVersionUID = -2929789292155268166L;

    @EqualsAndHashCode.Include
    private String bank;    //finCoNo
    @EqualsAndHashCode.Include
    private String finPrdtCd;

    public DepositPK(String finCoNo, String finPrdtCd) {
        bank = finCoNo;
        this.finPrdtCd = finPrdtCd;
    }
}

File: src/main/kotlin/com/hjj/apiserver/domain/deposit/DepositPK.kt
package com.hjj.apiserver.domain.deposit

import jakarta.persistence.Embeddable
import java.io.Serializable

@Embeddable
data class DepositPK(
    var bank: String,
    var finPrdtCd: String,
) : Serializable {
    companion object {
        private const val serialVersionUID: Long = 2379397187016502760L
    }
}
